numpy/core/src/umath/loops.c.src

```diff
@@ -840,28 +845,74 @@ NPY_NO_EXPORT NPY_GCC_OPT_3 void
     UNARY_LOOP_FAST(@type@, @type@, *out = in > 0 ? 1 : (in < 0 ? -1 : 0));
 }
 
+/* Libdivide only supports 32 and 64 bit types
+ * We try to pick the best possible one */
+/**begin repeat1
+ * #kind = t, gen, do#
+ */
+#if NPY_BITSOF_@TYPE@ <= 32
+#define libdivide_@type@_@kind@ libdivide_s32_@kind@
+#else
+#define libdivide_@type@_@kind@ libdivide_s64_@kind@
```

Comment:
I wonder if no repeat templating is just as well, but I don't care.

Reply:
I didn't quite get this point, can you please help me understand this.

Reply:
Oh like just declare at once instead of loop, I had a comment explaining [why](https://github.com/numpy/numpy/pull/17727/commits/3975a28d8b3efa385c58a0196f55d7d377e21a77#diff-0a2bc1c5f56bfc1ceb466e7c19faf20e215503e4f159b1bbf64a10125e78d2e5R849). Aim is, in future, I'll try some stuff like [Unswitching](https://libdivide.com/documentation.html#c_api) that will need a few more functions, that'll reduce code in long run

Reply:
yeah, it is just 6 lines of code, but a matter of opinion probably,

Reply:
Oh yeah, good point. I'll move them out of the nested loop.

Reply:
I think @seberg meant this?
```C
#if NPY_BITSOF_@TYPE@ <= 32
#define libdivide_@type@_t libdivide_s32_t
#define libdivide_@type@_gen libdivide_s32_gen
#define libdivide_@type@_do libdivide_s32_do
#else
#define libdivide_@type@_t libdivide_s64_t
#define libdivide_@type@_gen libdivide_s64_gen
#define libdivide_@type@_do libdivide_s64_do
#endif
```